Library/Homebrew/style.rb

```diff
@@ -149,13 +143,20 @@ def run_rubocop(files, output_type,
 
         args << "--color" if Tty.color?
 
-        system cache_env, RUBY_PATH, *ruby_args, *args
-        $CHILD_STATUS.success?
+        RuboCop::CLI.new.run(args.map(&:to_s)).zero?
       when :json
-        result = system_command RUBY_PATH,
-                                args: [*ruby_args, "--format", "json", *args],
-                                env:  cache_env
-        json = json_result!(result)
+        args = %w[--format json] + args
+
+        # Capture the output from rubocop's execution
```

Comment:
I wonder if there's any APIs that allow the JSON output to be obtained directly?

Reply:
Possibly. I've been looking through the code and slowly starting to wrap my head around it. I'd like to avoid using any of the private API (just seems like good practice) so that may be a limiting factor. So far, I know that `RuboCop::CLI#run` is part of the public API while other parts that may be helpful (like `RuboCop::Runner`) are private. I'll look into the JSON stuff, though.